### ISSUE TITLE
market API - price/amount given in list/bid not corresponding to create/bid

### ISSUE THREAD

**Issue by AkstonCap** [Nexusoft/LLL-TAO#165]
I'm getting some strange errors in market/create/bid and /ask, and also strange numbers in market/list/bid, /ask and /order for the price and amount which doesn't correspond to information given in create/bid and /ask. See attached pictures from API in 3.1.3 wallet (5.1.3-rc5 core version).

![marketAPI-240313-1](https://github.com/Nexusoft/LLL-TAO/assets/88211167/8502b498-766e-4279-bca4-7e6efba53a51)
![marketAPI-240313-2](https://github.com/Nexusoft/LLL-TAO/assets/88211167/188430fb-7323-44be-914d-fb9d36fe2bce)
![marketAPI-240313-3](https://github.com/Nexusoft/LLL-TAO/assets/88211167/431d82eb-daf4-46c5-a619-23bf4dec983f)

**Comment by AkstonCap** [comment]
Ah, okay.. I guess the "price" and first "amount" nr is given in all divisible units #NXS*1e6.. Now they make sense..
And I guess the error message comes from dist-token not having any decimals (bid for 0.5 tokens therefore not possible)..
Case closed then.

**Comment by AkstonCap** [comment]
Here's also an issue from create/bid (price is wrong):
![ksnip_20240317-130039](https://github.com/Nexusoft/LLL-TAO/assets/88211167/d6d90064-070a-4caf-bcb9-00aec11b3b22)

**Comment by AkstonCap** [comment]
Looks like create/ask doesn't allow NXS amounts that are not whole numbers. See images below; price=0.891 and token-amount=1 gave error, price=0.891 and token-amount=2 created ask successfully but returned as something else (price=0.5 and amount=2) in list/ask. This was done with core 5.1.3-rc5 (ask created in wallet 3.1.3, list/ask done in linux node).

![2024-03-24 - market_create_ask - 2](https://github.com/Nexusoft/LLL-TAO/assets/88211167/dcd76fd8-8e8d-499c-adda-82d85fd63c9f)
![2024-03-24 - market_create_ask](https://github.com/Nexusoft/LLL-TAO/assets/88211167/c8644e01-333e-42ac-a380-004ffd19e216)
![2024-03-24 - market_list_ask](https://github.com/Nexusoft/LLL-TAO/assets/88211167/8fcf11b4-30bd-42e1-93f0-2db621d01768)

**Comment by VidereLicet** [comment]
Yeah I'm going to integrate my precision_t class into the JSON library, IEEE754 standards don't hold values precisely and it's cased me so many headaches to catch all the edge cases.
